kernel/src/main/java/org/kframework/compile/GenerateSentencesFromConfigDecl.java

```diff
@@ -202,11 +209,31 @@ public void apply(KToken k) {
         }
 
         @Override
-        public void apply(KApply k) {
-            if (k.klabel().name().equals("#externalCell")) {
-                hasConfigVar = true;
+        public void apply(KApply kapp) {
+            if (kapp.klabel().name().equals("#externalCell")) {
+                if (kapp.klist().size() == 1) {
+                    K startLabel = kapp.klist().items().get(0);
+                    if (startLabel instanceof KToken) {
+                        KToken label = (KToken) startLabel;
+                        if (label.sort().equals(Sort("#CellName"))) {
+                            String cellName = label.s();
+                            Sort sort = Sort(getSortOfCell(cellName));
+                            Option<Set<Production>> initializerProduction = m.productionsFor().get(KLabel(getInitLabel(sort)));
+                            if (initializerProduction.isDefined()) {
+                                Set<Production> realProds = stream(initializerProduction.get())
+                                        .filter(p -> !p.att().contains(Att.RECORD_PRD(), Production.class))
+                                        .collect(Collections.toSet());
+                                if (realProds.size() == 1) { // should be only a single initializer
+                                    if (realProds.head().items().size() == 4) {
+                                      hasConfigVar = true;
+                                    }
+                                }
+                            }
```

Comment:
What is this change going on here? Is it related to the variable declaration issues?

Or is it related to the new issue @hjorthjort posted #1318.

Reply:
This is fixing a false positive that arose because we were generating a rule during parsing that had an unused variable.